src/ucp/proto/proto_common.c

```diff
@@ -150,14 +158,22 @@ size_t ucp_proto_common_get_iface_attr_field(const uct_iface_attr_t *iface_attr,
 static void
 ucp_proto_common_get_frag_size(const ucp_proto_common_init_params_t *params,
                                const uct_iface_attr_t *iface_attr,
-                               size_t *min_frag_p, size_t *max_frag_p)
+                               ucp_lane_index_t lane, size_t *min_frag_p,
+                               size_t *max_frag_p)
 {
     *min_frag_p = ucp_proto_common_get_iface_attr_field(iface_attr,
                                                         params->min_frag_offs,
                                                         0);
     *max_frag_p = ucp_proto_common_get_iface_attr_field(iface_attr,
                                                         params->max_frag_offs,
                                                         SIZE_MAX);
+
+    /* Adjust maximum fragment size taking into account segment size to prevent
+       sending more than the remote side supports. */
+    if (params->flags & UCP_PROTO_COMMON_INIT_FLAG_ADJ_MAX_FRAG) {
```

Comment:
@brminich are you sure this is better than checking send_op? 
seems its equivalent to send_op == am_xx || eager_xx

Reply:
Imo, it is better than using `uct_ep_operation_t`, like it was before (otherwise need to have 6 branches).
But do you mean using `UCP_OP_ID_TAG_SEND` and `UCP_OP_ID_AM_SEND`? What shall we do with amo/rma emulation and rndv replies then?

Reply:
branches should not be important here, it's slow-path. i means using UCT_EP_OP_xx